src/patches/dense_patch_computation.cpp

```diff
@@ -242,23 +322,70 @@ InstructionApplicationResult try_apply_instruction_direct_followup(
-                bell_init->loc2.target, ":", PauliOperator_to_string(bell_init->loc2.op))), {}};
+
+        if (!bell_init->counter)
+        {
+            auto routing_region = router.find_routing_ancilla(slice, bell_init->loc1.target, bell_init->loc1.op, bell_init->loc2.target, bell_init->loc2.op);
+            if(!routing_region) 
+            {
+                return {std::make_unique<std::runtime_error>(lstk::cat("No valid route found for Bell pair creation: ",
+                    bell_init->loc1.target, ":", PauliOperator_to_string(bell_init->loc1.op), ",",
+                    bell_init->loc2.target, ":", PauliOperator_to_string(bell_init->loc2.op))), {}};
+            }
+            else if (routing_region->cells.size() < 2) 
+            {
+                return {std::make_unique<std::runtime_error>(lstk::cat("Shortest route cannot be used for Bell pair creation: ",
+                    bell_init->loc1.target, ":", PauliOperator_to_string(bell_init->loc1.op), ",",
+                    bell_init->loc2.target, ":", PauliOperator_to_string(bell_init->loc2.op))), {}};
+            }
+
+            // Construct LocalInstructions based on route
+            std::vector<LocalInstruction> local_instructions;
+            for (size_t i=0; i<routing_region->cells.size()-1; i=i+2)
+            {
+                // Push a BellPrepare instruction with PatchID's depending on the case
+                if (i==0) 
+                    local_instructions.push_back({BellPrepare{bell_init->side2, std::nullopt, routing_region->cells[i].cell, routing_region->cells[i+1].cell}});
+                else if (i == routing_region->cells.size() - 2)
+                    local_instructions.push_back({BellPrepare{std::nullopt, bell_init->side1, routing_region->cells[i].cell, routing_region->cells[i+1].cell}});
+                else 
+                    local_instructions.push_back({BellPrepare{std::nullopt, std::nullopt, routing_region->cells[i].cell, routing_region->cells[i+1].cell}});
+            }
+            for (size_t i=2; i<routing_region->cells.size()-1; i=i+2)
+            {
+                // Push a complementary layer of BellMeasure instructions
+                local_instructions.push_back({BellMeasure{routing_region->cells[i-1].cell, routing_region->cells[i].cell}});
+            }
+            // Take care of the case of an odd route
+            if ((routing_region->cells.size()%2 == 1))
+            {
+                local_instructions.push_back({Move{bell_init->side1, routing_region->cells[routing_region->cells.size()-2].cell, routing_region->cells[routing_region->cells.size()-1].cell}});
+            }
+
+            // Update the LSInstruction itself
+            bell_init->local_instructions = local_instructions;
+            bell_init->counter = 0;
         }
 
-        apply_routing_region(slice, *routing_region);
+        // Apply LocalInstructions
+        if (bell_init->counter)
```

Comment:
This might be more intuitive as an else for the previous block. Also since `counter` is an optional int which are both implicitly convertible to bool. Might lead to some unexpected behavioir when `0`. Best to use has value here